### DISCUSSION TITLE
[FLUTTER] Appwrite OAuth2.0 api detects app link redirects as user canceling action by throwing PlatformException(User Canceled Login)

### DISCUSSION THREAD

**Discussion by devberkay** [appwrite/appwrite#3140] [Q&A]
this is my app link from manifest.xml : 

```
<intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data android:scheme="http" />
                <data android:scheme="https" />

                <data android:host="<domain>" />
            </intent-filter>
```

this is my googleOAuth function : 

```
Future<void> loginGoogleAuth() async {
    state = const AsyncLoading();
    Future result = _account.createOAuth2Session(
      success: "https://<domain>/googlevalid",
      failure: "https://<domain>/googleinvalid",
      provider: 'google',
    );

    result.then(
      (response) {
        print(response.runtimeType);
        state = const AsyncData(true);
        state = const AsyncData(null);
      },
    ).onError((e, st) {
      print('Error block has been executed');
      print(e);
      state = AsyncError(e!);
      _ref.refresh(authRequestStateProvider);
      state = const AsyncData(null);
    });
  }
```

this is my debug console logs after creating a new account & logging in with Google OAuth, which has contradictory results : 

```
I/flutter (26912): Received URI: https://<domain>/googlevalid
I/flutter (26912): Error block has been executed
I/flutter (26912): PlatformException(CANCELED, User canceled login, null, null)
```

Contradictory is that , it sends the success redirect link (and successfully create account or log in the account) but at the same time enters the .onError() block of code.

Note :` I/flutter (26912): Received URI: https://<domain>/googlevalid` this log come via usage of 'uni_links' plugin from pub.dev, 
which catches the app & deep links.

Problem is that , app links are designed to get back the app to the foreground automatically  by their nature. appwrite oauth api shouldn't detect it as a cancelling action that has come from user's gesture.

**Comment by stnguyen90** [comment]
For flutter mobile, the recommended approach is to leave the success and failure URLs blank. Appwrite will, then, automatically redirect the user back to the app based on the `appwrite-callback-[PROJECT_ID]` URL scheme. The only annoyance is a flash of a page saying "Missing Redirect URL" and that the page doesn't close automatically. There's [this](https://github.com/appwrite/appwrite/issues/3930) issue for that.
